src/test/kotlin/com/terraformation/backend/tracking/edit/PlantingSiteEditCalculatorTest.kt

```diff
@@ -0,0 +1,371 @@
+        "Number of permanent clusters to add when doubling zone size")
+    assertEquals(
+        PlantingZoneModel.DEFAULT_NUM_PERMANENT_CLUSTERS / 2,
+        calculateSiteEdit(existing, halfAgainSize)
+            .plantingZoneEdits
+            .first()
+            .numPermanentClustersToAdd,
+        "Number of permanent clusters to add when increasing zone size by half")
+    assertEquals(
+        1,
+        calculateSiteEdit(existing, slightlyLargerSize)
+            .plantingZoneEdits
+            .first()
+            .numPermanentClustersToAdd,
+        "Number of permanent clusters to add when increasing zone size slightly")
+  }
+
+  @Test
+  fun `does not add permanent cluster if there is not enough room for it in added area`() {
+    val existing = existingSite(width = 500)
+    val desired = newSite(width = 510)
+
+    assertEquals(
+        0,
+        calculateSiteEdit(existing, desired).plantingZoneEdits.first().numPermanentClustersToAdd,
+        "Number of permanent clusters to add")
+  }
+
+  @Test
+  fun `does not add permanent cluster if total usable area of zone has gone down`() {
+    val existing = existingSite(width = 10000, height = 500)
+    val desired =
+        newSite(width = 11000, height = 500) { exclusion = rectangle(width = 2000, height = 500) }
+
+    assertEquals(
+        0,
+        calculateSiteEdit(existing, desired).plantingZoneEdits.first().numPermanentClustersToAdd,
+        "Number of permanent clusters to add")
+  }
+
+  @Test
+  fun `removes monitoring plots that overlap with added exclusion area`() {
+    val existing = existingSite {
+      zone {
+        subzone {
+          plot()
+          plot()
+        }
+      }
+    }
+    val desired = newSite { exclusion = rectangle(1) }
+
+    assertEquals(
+        setOf(MonitoringPlotId(1)),
+        calculateSiteEdit(existing, desired)
+            .plantingZoneEdits
+            .firstOrNull()
+            ?.monitoringPlotsRemoved,
+        "Monitoring plots removed")
+  }
+
+  @Test
+  fun `removes monitoring plots that no longer lie in site`() {
+    val existing = existingSite {
+      zone {
+        subzone {
+          plot()
+          plot()
+        }
+      }
+    }
+    val desired = newSite(x = 1)
+
+    assertEquals(
+        setOf(MonitoringPlotId(1)),
+        calculateSiteEdit(existing, desired)
+            .plantingZoneEdits
+            .firstOrNull()
+            ?.monitoringPlotsRemoved,
+        "Monitoring plots removed")
+  }
+
+  @Test
+  fun `returns empty list of edits if nothing changed`() {
+    val site = existingSite()
+
+    assertEditResult(
+        site,
+        site.toNew(),
+        PlantingSiteEdit(
+            BigDecimal("0.0"), site.boundary!!, null, PlantingSiteId(1), emptyList(), emptyList()))
+  }
+
+  @Nested
+  inner class Validation {
+    @Test
+    fun `detects change to boundary between existing zones`() {
```

Comment:
Yes, this logic would only run if the site has plants. If it doesn't, my plan is to just delete the old map data entirely (saving it in history, of course) and create the new map since there's no need to preserve IDs and the fact that we allow arbitrary edits in that case means that the logic for matching old and new regions might fail even on valid edits.